em++

```diff
@@ -11,6 +11,9 @@
 # To make modifications to this file, edit `tools/run_python_compiler.sh` and
 # then run `tools/create_entry_points.py`
 
+ROOT=$(cd "$(dirname $0)" && pwd)
+export PYTHONPATH=$ROOT/.pip
```

Comment:
What happens if user wants to use their own system installed python, along with their own system installed pip plugins? Looks like this is unconditionally overriding/hiding user's own PYTHONPATH?

[PYTHONPATH supports multiple paths like PATH](https://docs.python.org/3/using/cmdline.html#envvar-PYTHONPATH). Maybe prepending $ROOT/.pip as first entry would work here?

Reply:
(also does LEB128 do anything else than that Wasm LEB128 encoding? If so, we could nuke using a separate module for that, and just write our own function.. [it's only a few lines](https://en.wikipedia.org/wiki/LEB128#Decode_unsigned_integer)?)

Reply:
> What happens if user wants to use their own system installed python, along with their own system installed pip plugins? Looks like this is unconditionally overriding/hiding user's own PYTHONPATH?
> 
> [PYTHONPATH supports multiple paths like PATH](https://docs.python.org/3/using/cmdline.html#envvar-PYTHONPATH). Maybe prepending $ROOT/.pip as first entry would work here?

Yes, it would be cleaner to add to the existing PYTHONPATH.

I did test with using system install modules though and this still works if you do `pip install -r requirements`.   This is because system python modules are found via PYTHONHOME which is always search in *addition* to PYTHONPATH, so that only time this wouldn't work would be if a user did something like this:

```
$ pip install -r requirements --target some/other/place  <-- installing emscripten dependencies somewhere random
$ export PYTHONPATH=some/other/place
```

In general we want emscripten to be has hermetic as possible so there might even be and argument made that we always want explicitly ignore the users PYTHONPATH (note: not their PYTHONHOME).   For example, a users PYTHONPATH might have user module in it that conflict with emscripten's module names.

But I digress, and I think I agree, its more polite to extent PYTHONPATH than to clobber it :)

Reply:
Done